### PR TITLE
Implement featured articles filter for article category/blog menu types.

### PR DESCRIPTION
**Pull request by roland-d**
Making this PR for @Klipper :)

In the module Articles category it is possible to select how featured articles are shown:
module_articles_category

This same filter is not available when making a menu item for a category list or category blog layout of articles. This PR implements this feature.

**Test instructions**
 After applying the PR a new option will show up in 2 places:
- Global configuration of articles under **_Shared options**_
  featured_articles_global_config
- Editing a menu item of the type **_Category Blog**_ and **_Category List**_ under **_List Layouts**_
  featured_articles_menu_config

This option works the same as all other options. The global article setting is used unless the setting is changed in the menu item as this will override the global setting.

There are 3 settings:
- Show
  This will show regular and featured articles
- Hide
  This will hide the featured articles
- Only
  This will show only the featured articles

Create a menu item of both the Category Blog and Category List type and test to see on the site if the correct articles are shown based on the settings chosen.

**Backwards compatibility**
As this is a new feature there are no backwards compatibility issues to expect. The option simply leverages a setting already available in a core module.

### CODE REVIEW

File: administrator/language/en-GB/en-GB.ini [hunk 1]
```diff
@@ -489,6 +490,8 @@ JGLOBAL_SHOW_EMAIL_ICON_DESC="Show/Hide the email icon. This allows you to email
 JGLOBAL_SHOW_EMAIL_ICON_LABEL="Show Email Icon"
 JGLOBAL_SHOW_EMPTY_CATEGORIES_DESC="If Show, empty categories will display. A category is only empty - if it has no items or subcategories."
 JGLOBAL_SHOW_EMPTY_CATEGORIES_LABEL="Empty Categories"
+JGLOBAL_SHOW_FEATURED_ARTICLES_DESC="Select to Show, Hide or Only display Featured Articles."
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.ini]: Can you change this to

+JGLOBAL_SHOW_FEATURED_ARTICLES_DESC="Select to show, hide or only display featured articles."

**Comment by brianteeman** [comment]
@test Works for me<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/5653">issues.joomla.org/joomla-cms/5653</a>.</sub>

**Comment by pe7er** [comment]
@test

In the front-end under "Article Category Blog" I have tested the display with "First Blog Post" and "Second Blog Post" and it performs as expected.

**Comment by zero-24** [comment]
Moving to RTC based on 2 successful tests. Thanks to all!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/5653">issues.joomla.org/joomla-cms/5653</a>.</sub>

File: administrator/language/en-GB/en-GB.ini [hunk 2]
```diff
@@ -489,6 +490,8 @@ JGLOBAL_SHOW_EMAIL_ICON_DESC="Show/Hide the email icon. This allows you to email
 JGLOBAL_SHOW_EMAIL_ICON_LABEL="Show Email Icon"
 JGLOBAL_SHOW_EMPTY_CATEGORIES_DESC="If Show, empty categories will display. A category is only empty - if it has no items or subcategories."
```
Comment (Hils) [inline administrator/language/en-GB/en-GB.ini]: A category is only empty - if it has no items...

Remove the hyphen?

Comment (brianteeman) [inline administrator/language/en-GB/en-GB.ini]: Good point Hils - As there are many instances of this sort of thing can you add it as an Issue on the en-GB issue queue and we can look at them all in one go

**Pull request merged**